hw1

zotmarya opened the pull request:
First homework.

zotmarya committed:
Added Task0 class

zotmarya committed:
Moved Task0 to package task0

Edited build status link in README.md file

zotmarya committed:
Added Task1 class.

Added Task1Test class with unit tests.

zotmarya committed:
Moved Task1Test class to task1 package.

zotmarya committed:
Added Task2 class.

Added Task2Test class with unit tests.

zotmarya committed:
Added Task3 class.

Added Task3Test class with unit tests.

zotmarya committed:
Renamed test3 package to task3.

zotmarya committed:
Added Task4 class.

Added Task4Test class with unit tests.

zotmarya committed:
Added Task5 class.

Added Task5Test class with unit tests.

zotmarya committed:
Added Task6 class.

Added Task6Test class with unit tests.

zotmarya committed:
Added Task7 class.

Added Task7Test class with unit tests.

zotmarya committed:
Added Task8 class.

Added Task8Test class with unit tests.

zotmarya committed:
Refactored unit tests by adding global variable for TaskN class object.

venommile commented:
Привет! Я буду проверять твои домашки) 
Немного правил и уточнений:
 Вот мой тг @mishagaltsev, можешь пинать меня с проверкой домашек/с развёрнутыми вопросами туда 2) Я тоже человек, могу что-то не так понять/не знать и т.д. Поэтому не бойся уточнять/спрашивать/спорить 3) Мои комментарии в МР’е будут делится на три типа/будут иметь три тега
 1 — `BUG` , это критичная ошибка/нарушение общепринятых конвенций, которую надо поправить 
2 — `ENHANCEMENT`, это НЕ ошибка, её можно не исправлять. Такой аналог «Я бы сделал по-другому». Их можно не исправлять, они влияют только на плюсики в карму) 
3 — `RESEARCH`, это просто ссылки лично от меня, чтобы расширить твой кругозор, показать как ещё можно было сделать то или иное. Это вообще никак не влияет на домашки, просто доп инфа, что можно почитать/посмотреть

venommile commented on src/main/java/edu/hw1/task1/Task1.java:
```diff
@@ -0,0 +1,25 @@
+
+            String[] numbersStr = videoTime.split(":");
+
+            return Integer.valueOf(numbersStr[0]) * minuteInSeconds + Integer.valueOf(numbersStr[1]);
```
`ENHANCEMENT` -- подумай, что будет если numbersStr[0] окажется очень большим, например Integer.MAX_VALUE

venommile commented on src/main/java/edu/hw1/task1/Task1.java:
```diff
@@ -0,0 +1,25 @@
+
+public class Task1 {
+
+    final String pattern = "[0-9]{2,}:[0-5]{1}[0-9]{1}";
```
`BUG` давай сделаем это констатной. константы в java называются в upper snake case `ВОТ_ТАК`

```suggestion
    static final String PATTERN = "[0-9]{2,}:[0-5]{1}[0-9]{1}";
```

venommile commented on src/main/java/edu/hw1/task2/Task2.java:
```diff
@@ -0,0 +1,12 @@
+
+    int digitsCount(int number) {
+
+        int absNumber = number < 0 ? -number : number;
```
`ENHANCEMENT` 
```suggestion
        int absNumber = Math.abs(number);
```

venommile commented on src/test/java/edu/hw1/task8/Task8Test.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    @Test
+    void checkIfTheyAreSafe() {
```
`BUG`  в джаве методы тестов принято называть в snake_case
```suggestion
    void check_if_they_are_safe() {
```

venommile commented on src/test/java/edu/hw1/task6/Task6Test.java:
```diff
@@ -0,0 +1,66 @@
+    }
+
+    @Test
+    void correctNumber() {
```
`RESEARCH` почитай про параметризованные тесты https://www.baeldung.com/parameterized-tests-junit-5

zotmarya committed:
Fixed constant variables' names using uppercase with underscore to follow naming convention in Task1 class.

zotmarya committed:
Changed usage of ternary operator to Math.abs() method in Task2 class.

zotmarya committed:
Fixed constant variables' names using upper snake case to follow naming convention Task5 class.

zotmarya committed:
Fixed constant variables' names using upper snake case to follow naming convention in Task6 class.

Added some code in stepsCountForK() method to add a leading zero if necessary.

zotmarya committed:
Fixed constant variables' names using uppercase with underscore to follow naming convention in Task7 class.

zotmarya committed:
Fixed constant variables' names using uppercase with underscore to follow naming convention in Task8 class.

zotmarya committed:
Added constant variable DIGITS_IN_NUMBER for array size in Task6 class.

zotmarya committed:
Fixed array out of bounds for index j by subtracting 1 from array length in Task6 class.

zotmarya committed:
Changed an access modifier for method isOnBoard from public to private in Task8 class.

zotmarya committed:
Changed test methods' names to follow convention style for Task1Test-Task8Test classes.

zotmarya committed:
Added parametrized tests in Task1Test, Task2Test, Task4Test-Task7Test classes.

Added new test for when there are no knights on the chessboard in Task8Test class.
